extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -114,6 +131,43 @@ public void initialize() {
         }, 0, 1, TimeUnit.MINUTES);
     }
 
+    protected GatewayConfig getUsernameAndKey(GatewayConfig configuration) {
```

Comment:
The method name is pretty distracting here. One would never expect calls to the gateway and especially no side effects like updating the Thing configuration,

Reply:
Could you please suggest better name?
(How about ```authenticateOnGateway``` or something similar ?)

Reply:
You should probably split it into at least two methods: One that takes an IDENTITY and provides a PSK and another that updates the Configuration as needed.